src/status_im2/contexts/chat/bottom_sheet_composer/effects.cljs

```diff
@@ -0,0 +1,91 @@
+(ns status-im2.contexts.chat.bottom-sheet-composer.effects
+  (:require
+    [react-native.async-storage :as async-storage]
+    [react-native.core :as rn]
+    [react-native.reanimated :as reanimated]
+    [status-im2.contexts.chat.bottom-sheet-composer.constants :as c]
+    [status-im2.contexts.chat.bottom-sheet-composer.keyboard :as kb]
+    [utils.re-frame :as rf]))
+
+(defn reenter-screen-effect
+  [{:keys [text-value saved-cursor-position maximized?]}
+   {:keys [content-height]}
+   {:keys [input-content-height input-text input-maximized?]}]
+  (when (and (empty? @text-value) (not= input-text nil))
+    (reset! text-value input-text)
+    (reset! content-height input-content-height)
+    (reset! saved-cursor-position (count input-text)))
+  (when input-maximized?
+    (reset! maximized? true)))
+
+(defn maximized-effect
+  [{:keys [maximized?]}
+   {:keys [height saved-height last-height]}
+   {:keys [max-height]}
+   {:keys [input-content-height]}]
+  (when (or @maximized? (>= input-content-height max-height))
+    (reanimated/animate height max-height)
+    (reanimated/set-shared-value saved-height max-height)
+    (reanimated/set-shared-value last-height max-height)))
+
+(defn refocus-effect
+  [{:keys [input-ref]}
+   {:keys [input-refocus?]}]
+  (when input-refocus?
+    (.focus ^js @input-ref)
+    (rf/dispatch [:chat.ui/set-input-refocus false])))
+
+(defn layout-effect
+  [{:keys [lock-layout?]}]
+  (when-not @lock-layout?
+    (js/setTimeout #(reset! lock-layout? true) 500)))
+
+(defn kb-default-height-effect
+  [{:keys [kb-default-height]}]
+  (when-not @kb-default-height
+    (async-storage/get-item :kb-default-height
```

Comment:
Why do we need to store the height and then read it?

Usually these operations (reading from local storage) are performed while the app is initializing.

Would it be a good addition to add a loader while the data is being read?
Do you think moving this to the app initialization and store it in re-frame is a good option?

Reply:
Yes, I should move it to one of the earlier screens. I'll do so once the onboarding implementation is complete. 

I need to store the keyboard-height in `async-storage` and not `re-frame` because I need to know the KB height before opening it (for some calculations). Ofc, in `re-frame` the data won't be persisted.

Reply:
oh, sorry for not being clear, I meant, reading from local storage in app initialization and store the data read in re-frame, so you read the KB height from re-frame.

Ok, let's wait for onboarding to be ready to move this :+1: